src/framework/core/js/DataBinding.js

```diff
@@ -1507,4 +1507,76 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         return that;
     };
 
+    /**
+     * Calculates the changes between the model values 'value' and
+     * 'oldValue'. The optional argument 'changePathPrefix' is prepended
+     * to the generated change paths. This is useful for generating
+     * change records to be applied at a non-root path in a model.
+     * Returns an array of change records, suitable for passing to
+     * fluid.fireChanges().
+     */
+    fluid.modelPairToChanges = function (value, oldValue, changePathPrefix) {
+        changePathPrefix = changePathPrefix || "";
+
+        var diffOptions = {changes: 0, unchanged: 0, changeMap: {}};
+        fluid.model.diff(oldValue, value, diffOptions);
+
+        var changes = [];
+
+        fluid.modelPairToChanges1(
+            value,
+            fluid.pathUtil.parseEL(changePathPrefix),
+            diffOptions.changeMap,
+            [],
+            changes
+        );
+
+        return changes;
+    };
+
+    fluid.modelPairToChanges1 = function (value, changePathPrefixSegs, changeMap, changeSegs, changes) {
+        if (changeMap === "ADD") {
+            // The whole model value is new
+            changes.push({
+                path: changePathPrefixSegs,
+                value: value,
+                type: "ADD"
+            });
+        } else if (changeMap === "DELETE") {
+            // The whole model value has been deleted
+            changes.push({
+                path: changePathPrefixSegs,
+                value: null,
+                type: "DELETE"
+            });
+        } else if (fluid.isPlainObject(changeMap, true)) {
+            // Something within the model value has changed
+            fluid.each(changeMap, function (change, seg) {
+                var currentChangeSegs = changeSegs.concat([seg]);
+                if (change === "ADD") {
+                    changes.push({
+                        path: changePathPrefixSegs.concat(currentChangeSegs),
+                        value: fluid.get(value, currentChangeSegs),
+                        type: "ADD"
+                    });
+                } else if (change === "DELETE") {
+                    changes.push({
+                        path: changePathPrefixSegs.concat(currentChangeSegs),
+                        value: null,
+                        type: "DELETE"
+                    });
+                } else if (fluid.isPlainObject(change, true)) {
+                    // Recurse down the tree of changes
+                    fluid.modelPairToChanges1(
```

Comment:
We could probably get a few of these arguments onto the same line? I'm fully supportive of our new shorter line length but still keen to see vertical space used as sparingly as we can given that :)